src/ioc.ts

```diff
@@ -217,6 +210,13 @@ export const createGlobalContainer = (options: {
   container.bind(Execa).toConstantValue(execa);
   container.bind(FS).toConstantValue(fsPromises);
 
+  // Core services:
+  container
+    .bind(ILogger)
+    .to(Logger)
+    .inSingletonScope()
+    .onActivation(trackDispose);
```

Comment:
The logger is configured by and bound to the parameters in a single session; I don't think it should be created at a global level.

Reply:
I suppose we can manually inject the logger in to the transport when we create the top level session, does that work for you?

Reply:
Alternately, maybe the transport could not need a logger and the DAP Connection could do all the necessary logging?

Reply:
considered that, but I like keeping it in the transport since that's where we're adding the sessionId for flat session, and if we move it up to connection we won't get the sessionId in logs